sklearn/base.py

```diff
@@ -290,10 +290,15 @@ def __repr__(self):
                                                offset=len(class_name),),)
 
     def __getstate__(self):
+        try:
+            state = super(BaseEstimator, self).__getstate__()
+        except AttributeError:
+            state = self.__dict__
+
         if type(self).__module__.startswith('sklearn.'):
```

Comment:
My idea was to write something like this

```
    def test_multiple_inheritance_setting_foreign_namespace(self):
        try:
            estimator = MultiInheritanceEstimator()
            old_mod = type(estimator).__module__
            type(estimator).__module__ = "notsklearn"

            serialized = pickle.dumps(estimator, protocol=2)
        finally:
            type(estimator).__module__ = old_mod
```
which doesn't work for the aforementioned reason.